modules/phase_field/src/kernels/KKSACBulkF.C

```diff
@@ -44,70 +34,68 @@ KKSACBulkF::KKSACBulkF(const InputParameters & parameters)
   : KKSACBulkBase(parameters),
     _c_map(getParameterJvarMap("global_cs")),
     _num_c(coupledComponents("global_cs")),
+    _c_names(coupledComponents("global_cs")),
     _ci_names(getParam<std::vector<MaterialPropertyName>>("ci_names")),
-    _dcidb_names(getParam<std::vector<MaterialPropertyName>>("dcidb_names")),
-    _prop_dcidb(_num_c),
-    _dcideta_names(getParam<std::vector<MaterialPropertyName>>("dcideta_names")),
-    _prop_dcideta(_num_c),
     _Fa_name(getParam<MaterialPropertyName>("fa_name")),
     _first_dFa(_num_c),
     _Fb_name(getParam<MaterialPropertyName>("fb_name")),
-    _prop_Fb(getMaterialProperty<Real>("fb_name")),
     _first_dFb(_num_c),
     _prop_dg(getMaterialPropertyDerivative<Real>("g_name", _eta_name)),
     _prop_d2g(getMaterialPropertyDerivative<Real>("g_name", _eta_name, _eta_name)),
-    _L(getMaterialProperty<Real>("L_name")),
-    _w(getParam<Real>("w"))
+    _w(getParam<Real>("w")),
+    _prop_Fi(2),
+    _dcideta(_num_c * 2),
+    _dcidb(_num_c * _num_c * 2)
 {
+  _prop_Fi[0] = &getMaterialPropertyByName<Real>("cp" + _Fa_name);
+  _prop_Fi[1] = &getMaterialPropertyByName<Real>("cp" + _Fb_name);
```

Comment:
@dschwen In this push I added the energy Fi and derivative (dFidci and dcidc) calculations in the materials with names generated automatically, and updated the kernels. Is this how we grab the Fi from the materials? dcidc seems to be zero in the kernels even though it was declared in KKSPhaseConcentrationDerivatives. Also the pull from moose outdates my compiler so I can't compile to test right now, hopefully it is easy to fix. (I will give the kernels new names by the end.)